File: client/src/app/model/product.ts
export interface Product {
    id: number;
    name: string;
    description: string;
    price: number;
    pictureUrl: string;
    type?: string;
    author: string;
    quantityInStock?: number;
    rating?: number;
}
export interface ProductParams {
    orderBy: string;
    searchTerm?: string;
    types: string[];
    brands: string[];
    pageNumber: number;
    pageSize: number;
}

File: BookStoreOnl/Entities/Product.cs
using System.ComponentModel.DataAnnotations;

namespace BookStoreOnl.Entities
{
    public class Product
    {
        [Key]
        public int Id { get; set; }
        public string Name { get; set; }
        public string Description { get; set; }
        public long Price { get; set; }
        public string PictureUrl { get; set; }
        public string Type { get; set; }
        public string Author { get; set; }
        public int QuantityInStock { get; set; }
        public int Rating { get; set; }
        public string PublicId { get; set; }
    }
}
